src/frontend/screens/Observation/Buttons.tsx

```diff
@@ -104,11 +104,16 @@ export const ButtonFields = ({
   async function handlePressShare() {
     const {lon, lat} = observation;
     setShareButtonLoading(true);
-
-    const urlsQueries = await Promise.all(
-      attachmentUrlQueries.map(q => q.refetch()),
-    );
-    const urls = urlsQueries.map(query => query.data!.url);
+    const needsRefetch = attachmentUrlQueries.some(query => !query.data);
+    let urls;
+    if (needsRefetch) {
+      const urlsQueries = await Promise.all(
+        attachmentUrlQueries.map(q => q.refetch()),
+      );
+      urls = urlsQueries.map(query => query.data!.url);
+    } else {
+      urls = attachmentUrlQueries.map(query => query.data!.url);
+    }
```

Comment:
starting to question the need of using react-query (specifically `useAttachmentUrlQueries()`) for this data as opposed to just calling the project api directly. Using the query would make sense if the query state needed to show up or affect what's rendered in the UI, but in this case, we're mostly interested in getting the urls on demand based on a specific user interaction. Thoughts on doing the following instead of using that rather complex query?

```ts
// Make sure this is set up in the component
const {projectApi} = useActiveProject()

// Then handlePressShare()
async function handlePressShare() {
  const {lon, lat} = observation;
  setShareButtonLoading(true);

  // Using `Promise.allSettled()` but depending on UX, may want to use `Promise.all()`
  // If using the latter, should make sure to handle the case if one of the calls throws an error
  const urlResults = await Promise.allSettled(
    observation.attachments.map(a => {
      if (
        a.type === 'UNRECOGNIZED' ||
        a.type === 'attachment_type_unspecified'
      ) {
        return null;
      } else
        return projectApi.$blobs.getUrl({
          variant: 'original',
          type: a.type,
          name: a.name,
          driveId: a.driveDiscoveryId,
        });
    }),
  );

  const validUrls = urlResults
    .filter(
      (result): result is PromiseFulfilledResult<string> =>
        result.status === 'fulfilled' && result.status !== null,
    )
    .map(result => result.value);

  const base64Urls = await Promise.all(
    validUrls.map(url => convertUrlToBase64(url)),
  );
```

Reply:
hm actually on second thought, this results in a different UX because now we're always fetching whenever the button is pressed, as opposed to using a cache. Could adapt this to hold some state in this component, but then we're probably just doing a lazy version of react query in that case.

Maybe my suggestion isn't worth considering after all 😅